pycbc/waveform/bank.py

```diff
@@ -381,10 +378,12 @@ def write_to_hdf(self, filename, start_index=None, stop_index=None,
         write_tbl = self.table[start_index:stop_index]
         for p in parameters:
             f[p] = write_tbl[p]
-        if self.compressed_waveforms is not None:
+        if self.check_compressed_waveforms:
             for tmplt_hash in write_tbl.template_hash:
-                self.compressed_waveforms[tmplt_hash].write_to_hdf(
-                                                        f, tmplt_hash)
+                compressed_waveform = pycbc.waveform.compress.CompressedWaveform.from_hdf(
+                                        self.filehandler, tmplt_hash,
+                                        load_now=self.load_compressed_now)
```

Comment:
Just set `load_now=True`. You have to load the sample points now, since they are going to be used immediately.